stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -74,6 +84,19 @@ public struct ${Self}<Pointee>
     _rawValue = from._rawValue
   }
 
+  /// Convert from any `UnsafeMutablePointer`, possibly with a
+  /// different `Pointee`.
+  ///
+  /// Returns nil if `from` is nil.
+  ///
+  /// - Warning: the behavior of accesses to pointee as a type
+  ///   different from that to which it was initialized is undefined.
+  @_transparent
+  public init?<U>(_ from : UnsafeMutablePointer<U>?) {
+    guard let unwrapped = from else { return nil }
```

Comment:
Please add tests for new APIs and tests for new behavior in existing initializers.